tcp-native/src/main/java/org/ballerinalang/stdlib/tcp/nativeclient/Client.java

```diff
@@ -57,7 +58,7 @@ public static Object externInit(Environment env, BObject client, BString remoteH
             localAddress = new InetSocketAddress(host.getValue(), 0);
         }
 
-        long timeout = config.getIntValue(StringUtils.fromString(Constants.CONFIG_READ_TIMEOUT));
+        double timeout = ((BDecimal) config.get(StringUtils.fromString(Constants.CONFIG_READ_TIMEOUT))).floatValue();
```

Comment:
Can't we cast like this?
```suggestion
        long timeout = ((BDecimal) config.get(StringUtils.fromString(Constants.CONFIG_READ_TIMEOUT))).decimalValue().longValue();
```

Reply:
if the user needs to give timeout as 100ms then user will give it as timeout=0.1, if we cast this to long we will lose the precision.